toolkit/scss/forms/_keyword-search.scss

```diff
@@ -2,10 +2,83 @@
 @import "_css3.scss";
 @import "_colours.scss";
 
-.govuk-option-select .filter-field-text {
-  @include core-16;
-  @include box-sizing(border-box);
-  width: 100%;
+
+.govuk-option-select.filter-field-text {
+
+  margin-bottom: $gutter;
   padding: 5px;
-  border: 1px solid $border-colour;
+
+  .js-enabled & {
+      /* This is required because we want padding around the search box but it gets overridden by a style from _option_select.scss */
+      padding: 5px;
+  }
+
+  .container-head {
+    padding-left: 0;
+    padding-right: 0;
+  }
+
+  .option-select-label {
+    display: block;
+  }
+
+  .container-search {
+    position: relative;
+  }
+
+  input {
+    height: 36px;
+  }
+
+  /* Mostly gotten from: https://github.com/alphagov/static/blob/bb2e0b052c15e5cfc58a0dca3a34bd4cc5caed46/app/assets/stylesheets/helpers/_header.scss#L102 */
+
+  input[type="search"] {
+    @include core-16;
+    @include box-sizing(border-box);
+    position: relative;
+    padding: 6px;
+    width: 86%;
+    border: 1px solid $border-colour;
+    border-right: none;
+    background: #fff;
+
+    &.focus,
+    &:focus {
+      outline-offset: -2px;
+    }
+
+    .js-enabled & {
+      @include calc(width, "100% - 36px");
+    }
```

Comment:
Actually, in light of your earlier question about what `.js-enabled` is actually doing for us, I don't think we need this. 

Presumably, even without javascript running, we still want the `calc` rule to work on browsers that support it.  I think you're probably fine to get rid of this `.js-enabled &` rule and move the calc declaration just after the `width` rule above.